addons/mail/static/src/core/common/composer.js

```diff
@@ -685,6 +686,12 @@ export class Composer extends Component {
         return markup`<div>${defaultBody}</div>`; // as to not wrap in <p> by html_sanitize
     }
 
+    handleValidPartnerMention(partnerLinks) {
+        this.props.composer.mentionedPartners = partnerLinks.map((el) =>
+            this.store["res.partner"].get(Number(el.dataset.oeId))
+        );
```

Comment:
The target partner might not be locally known in the current JS store (`get()` failing) but it could still be a potentially valid one.

If `get()` is failing this should do RPC, internal users might be able to find the partner with ACL, but for portal/guest we also need to provide the mention token to validate the access.

This means the mention token should be saved/restored as well.

Reply:
maybe I should introduce `getOrFetch` in res partner as well?

Reply:
with the `fetch` in the checker, we dont have to do the fetch here since the record is guaranteed in the store?